superset/db_engine_specs/hive.py

```diff
@@ -429,12 +429,26 @@ def execute(  # type: ignore
 
     @classmethod
     @cache.memoize()
-    def get_function_names(cls, database: "Database") -> List[str]:
+    def _get_function_names(cls, database: "Database") -> List[str]:
```

Comment:
@villebro the `_get_function_names` function could be nested within the `get_function_names` method if you like.

Reply:
Very good point. Somehow I instinctively wanted to avoid the overhead of regenerating the nested function, but seems I grossly overestimated the cost: "[...]it takes less time than creating an empty dict" https://stackoverflow.com/questions/7839632/is-there-an-overhead-when-nesting-functions-in-python